### PR TITLE
feat(converter): add Reference Object summary and description plugins

### PR DESCRIPTION
**Pull request by tekyu**
his plugin removes the `summary` and `description` fields of `Reference Object`.

Refs SWG-10235
Refs #3697

### CODE REVIEW

File: packages/apidom-converter/src/strategies/openapi-3-1-to-openapi-3-0-3/refractor-plugins/reference-summary-description.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        ReferenceElement(element: ReferenceElement) {
+          if (element.hasKey('summary')) {
+            annotations.push(summaryAnnotation);
+            element.remove('summary');
+          }
+          if (element.hasKey('description')) {
+            annotations.push(descriptionAnnotation);
+            element.remove('description');
+          }
+          return undefined;
```
Comment (char0n) [inline packages/apidom-converter/src/strategies/openapi-3-1-to-openapi-3-0-3/refractor-plugins/reference-summary-description.ts:33]: No need to return here. the `return` will be implicit and equivalent if we remove this code.

```suggestion
          return undefined;
```

**Comment by char0n** [comment]
@tekyu there is an option to have this plugin divided into two. One plugin that removes summary, one that removes description. The idea is that every plugin just handles one aspect of the the transformation.

In this particular case, we can say that the aspect is to remove the non-supported fields, or we can divide the aspect into two separate ones. 

Please let me know what you think.

**Comment by tekyu** [comment]
@char0n if the idea is to have plugins that handles single operation I don't want to disrupt that with my plugin. I'll split them then

**Comment by char0n** [comment]
@tekyu it's your decision, as for this particular one I don't have strong opinion.

File: packages/apidom-converter/src/strategies/openapi-3-1-to-openapi-3-0-3/refractor-plugins/reference-description.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import { AnnotationElement } from '@swagger-api/apidom-core';
+
+type ReferenceDescriptionPluginOptions = {
+  annotations: AnnotationElement[];
+};
+
+const referenceDescriptionRefractorPlugin =
+  ({ annotations }: ReferenceDescriptionPluginOptions) =>
+  () => {
+    const descriptionAnnotation = new AnnotationElement(
```
Comment (char0n) [inline packages/apidom-converter/src/strategies/openapi-3-1-to-openapi-3-0-3/refractor-plugins/reference-description.ts:11]:
```suggestion
    const annotation = new AnnotationElement(
```

File: packages/apidom-converter/src/strategies/openapi-3-1-to-openapi-3-0-3/refractor-plugins/reference-summary.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import { AnnotationElement } from '@swagger-api/apidom-core';
+
+type ReferenceSummaryPluginOptions = {
+  annotations: AnnotationElement[];
+};
+
+const referenceSummaryRefractorPlugin =
+  ({ annotations }: ReferenceSummaryPluginOptions) =>
+  () => {
+    const summaryAnnotation = new AnnotationElement(
```
Comment (char0n) [inline packages/apidom-converter/src/strategies/openapi-3-1-to-openapi-3-0-3/refractor-plugins/reference-summary.ts:11]:
```suggestion
    const annotation = new AnnotationElement(
```

**Review by char0n — APPROVED** [review]
Two nitpicks related to naming, otherwise LGTM

**Comment by char0n** [comment]
@tekyu plese use `Squash + merge` github feature described in https://github.com/swagger-api/apidom/pull/3848#issuecomment-1959582561


Please also use commit message format described in: https://github.com/swagger-api/apidom/pull/3848#issuecomment-1959505764

**Pull request merged**